README.md

```diff
@@ -1,2 +1,188 @@
-# spacelift-cli
-Spacelift client and CLI
+# Spacelift CLI
+
+Spacelift CLI (`spacelift-cli`) is a utility wrapping Spacelift's [GraphQL API](https://docs.spacelift.io/integrations/api) for easy programmatic access in command-line contexts - either in manual interactive mode (in your local shell), or in a predefined CI pipeline (GitHub actions, CircleCI, Jenkins etc).
+
+## Installation
+
+Spacelift CLI is distributed through GitHub Releases as a gzipped tarball containing a self-contained statically linked executable built from the source in this repository. Binaries can be download directly from the [Releases page](https://github.com/spacelift-io/spacelift-cli/releases) or programmatically using `curl` or `wget`. The example below covers macOS (Intel CPU) installation of the latest available release. In order to download and install a Linux version, change `darwin` to `linux`. If you're using an ARM processor (eg. Raspberry Pi, Apple Silicon), change `amd64` to `arm64`:
+
+```bash
+curl -s -L https://github.com/spacelift-io/spacelift-cli/releases/latest/download/spacelift-cli-darwin-amd64.tar.gz | tar xz -C /tmp && \
+mv /tmp/spacelift-cli-darwin-amd64 /usr/local/bin/spacelift-cli
+```
+
+## Authentication
+
+Spacelift CLI is designed to work in two different contexts - a non-interactive scripting mode (eg. external CI/CD pipeline) and local interactive mode, meaning you typing things in your shell. It thus supports two types of credentials - environment variables and user profiles. Let's call these "credential providers", like AWS does. We'll look into them separately.
+
+### Authenticating using environment variables
+
+When certain variables are defined in the environment, `spacelift-cli` will try to use them as Spacelift API credentials. First, the `SPACELIFT_API_TOKEN`. IF this variable is present in the environment, it alone contains enough information to set up a Spacelift session. Hence, the CLI will look no further and attempt to use it to talk to the server. Note that these tokens are generally short-lived.
+
+If the `SPACELIFT_API_TOKEN` variable is not present, the CLI will look for the
+`SPACELIFT_API_ENDPOINT` variable. This one contains the API endpoint to talk to, essentially pointing to your account and (optionally) Spacelift server - for example, `https://mycorp.app.spacelift.io`. If this variable is not present, environment lookup fails. Otherwise, it proceeds to credentials lookup.
```

Comment:
Sure. It means a different API server. All the vanity subdomains in `app.spacelift.io` end up hitting the same load balancer, but our preprod is in `app.spacelift.dev` and if/once we spin up the US region, the API server for that may be `app.spacelift.us`. Similar for on-prems, if we ever have them. So this is pure future-proofing.

Reply:
Cool - that makes sense. I can't quite make my mind up here, but it might be worth just removing the `(optionally)` on the grounds that it's not optional (i.e. you always need to provide the `app.spacelift.io` part).